### PR TITLE
Add 'otherApps' capability

### PR DESCRIPTION
**Pull request by dpgraham**
* Augmented initAUT to install apk's if a JSON array or string was provided in capability called 'otherApps'
* Goes through the list and downloads (if URL) the apps asynchronously and then installs all of them asynchronously using ADB install

(next step will be to document in caps.md)

### CODE REVIEW

File: lib/driver.js [hunk 1]
```diff
@@ -462,6 +472,23 @@ class AndroidDriver extends BaseDriver {
     // this will change depending on ChromeDriver status
     return _.isFunction(this.proxyReqRes);
   }
+
+  parseArray (cap) {
```
Comment (mykola-mokhnach) [inline lib/driver.js]: I'd rather move this helper to androidHelpers

File: lib/android-helpers.js [hunk 2]
```diff
@@ -301,6 +303,25 @@ helpers.installApk = async function (adb, opts = {}) {
...
+helpers.installOtherApks = async function (otherApps, adb, opts) {
+  let {androidInstallTimeout = PACKAGE_INSTALL_TIMEOUT,
+    autoGrantPermissions} = opts;
+
+  // Install all of the APK's asynchronously
+  await B.all(otherApps.map((otherApp) => {
+    logger.debug(`Installing app: ${otherApp}`);
+    return adb.install(otherApp, {
+      grantPermissions: autoGrantPermissions,
+      timeout: androidInstallTimeout
```
Comment (mykola-mokhnach) [inline lib/android-helpers.js]: should we also consider upgrade option (`install -r`)?

Comment (dpgraham) [reply lib/android-helpers.js]: Good call. We should always replace the apps.

**Comment by dpgraham** [comment]
Fixes pushed

File: lib/android-helpers.js [hunk 3]
```diff
@@ -301,6 +303,26 @@ helpers.installApk = async function (adb, opts = {}) {
...
+  let {androidInstallTimeout = PACKAGE_INSTALL_TIMEOUT,
+    autoGrantPermissions} = opts;
+
+  // Install all of the APK's asynchronously
+  await B.all(otherApps.map((otherApp) => {
+    logger.debug(`Installing app: ${otherApp}`);
+    return adb.install(otherApp, {
+      grantPermissions: autoGrantPermissions,
+      timeout: androidInstallTimeout,
+      "-r": true,
```
Comment (mykola-mokhnach) [inline lib/android-helpers.js]: I don't think this is the correct option name :)
Use `replace: true`

File: lib/android-helpers.js [hunk 4]
```diff
@@ -562,6 +584,26 @@ helpers.removeAllSessionWebSocketHandlers = async function (server, sessionId) {
...
+  // If it's a string, try parsing as an object/array
+  if (_.isString(cap)) {
+    let parsedCaps;
+    try {
+      parsedCaps = JSON.parse(cap);
+    } catch (ign) { }
+
+    if (_.isArray(parsedCaps)) {
+      return parsedCaps;
+    } else if (_.isPlainObject(parsedCaps)) {
```
Comment (mykola-mokhnach) [inline lib/android-helpers.js]: I'd rather check like this:

```
if (_.isArray(parsedCaps)) {
      return parsedCaps;
}

if (_.isString(parsedCaps)) {
  return [parsedCaps];
}

throw new Error....
```

File: lib/android-helpers.js [hunk 5]
```diff
@@ -562,6 +584,26 @@ helpers.removeAllSessionWebSocketHandlers = async function (server, sessionId) {
   }
 };
 
+helpers.parseArray =  function (cap) {
```
Comment (mykola-mokhnach) [inline lib/android-helpers.js]: extra space here

File: test/unit/android-helper-specs.js [hunk 6]
```diff
@@ -392,6 +392,38 @@ describe('Android Helpers', function () {
       mocks.helpers.verify();
     });
   }));
+  describe('installOtherApks', withMocks({adb, fs, helpers}, function (mocks) {
+    const opts = {
+      app : 'local',
```
Comment (imurchie) [inline test/unit/android-helper-specs.js]: extra spaces after property name, before colon.

File: test/unit/android-helper-specs.js [hunk 7]
```diff
@@ -655,4 +687,17 @@ describe('Android Helpers', function () {
...
+  describe('#parseArray', function () {
+    it('should parse array string to array', function () {
+      helpers.parseArray('["a", "b", "c"]').should.eql(['a', 'b', 'c']);
+    });
+    it('should parse a simple string to one item array', function () {
+      helpers.parseArray('abc').should.eql(['abc']);
+    });
+    it('should reject if an Object is provided', function () {
+      (() => helpers.parseArray('{"hello": "world"}')).should.throw(/s/);
+      //(() => helpers.parseArray('{"hello": "world"}'))().should.throw(/must provide a/);
```
Comment (imurchie) [inline test/unit/android-helper-specs.js]: ?

**Comment by dpgraham** [comment]
Changes pushed

**Comment by dpgraham** [comment]
Published to 2.23.0

**Pull request merged**